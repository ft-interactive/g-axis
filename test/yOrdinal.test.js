/**
 * @file
 * Test suite for xDate
 */
import { toMatchImageSnapshot } from 'jest-image-snapshot';
import listen from 'test-listen';
import micro from 'micro';
import { launch } from 'puppeteer';
import { rollup } from 'rollup';

expect.extend({ toMatchImageSnapshot });

let server;
let url;
let code;

beforeAll(async () => {
    const bundle = await rollup({
        input: `${__dirname}/../src/yOrdinal.js`,
        external: [
            'd3',
        ],
    });

    const output = await bundle.generate({
        format: 'umd',
        name: 'yOrdinal',
        globals: {
            d3: 'd3',
        },
    });

    code = output.code;
});

beforeEach(async () => {
    server = micro(async () => `<!doctype html>
    <html>
    <head>
    <script src="http://cdnjs.cloudflare.com/ajax/libs/d3/4.13.0/d3.js"></script>
    <script src="https://unpkg.com/g-chartframe"></script>
    <script>${code}</script>
    </head>
    <body><svg /></body>
    </html>`);
    url = await listen(server);
});

afterEach(() => server.close());

test('yOrdinal()', async () => {
    const browser = await launch();
    const page = await browser.newPage();
    await page.goto(url);

    // Add xDate
    await page.addScriptTag({
        content: code,
    });

    await page.evaluate(async () => {
        const sharedConfig = {
            source: 'Source not yet added',
            subtitle: 'Subtitle not yet added',
            title: 'Title not yet added',
        };
        const svg = await window.d3.select(document.querySelector('svg'));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig)
            .margin({
                bottom: 86,
                left: 20,
                right: 5,
                top: 100,
            })
            // .title("Put headline here")
            .height(500);

        // Instantiate the chart frame
        svg.call(currentFrame);
        const myScale = window.yOrdinal();
        currentFrame.plot().call(myScale);
    });

    const image = await page.screenshot();
    await browser.close();

    expect(image).toMatchImageSnapshot();
});
