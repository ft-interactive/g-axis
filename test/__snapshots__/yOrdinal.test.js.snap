// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`left-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis yAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\">
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,32.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"-3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Oranges</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,103.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"-3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Lemons</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,174.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"-3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Apples</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,245.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"-3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Pears</text>
  </g>
</g>"
`;

exports[`right-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis yAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"start\\" transform=\\"translate(542, 0)\\">
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,32.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Oranges</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,103.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Lemons</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,174.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Apples</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,245.5)\\">
    <line stroke=\\"#000\\" x2=\\"0\\"></line><text fill=\\"#000\\" x=\\"3\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">Pears</text>
  </g>
</g>"
`;
